silk/tests/openthread/ot_test_address_cache_table.py

```diff
@@ -0,0 +1,435 @@
+
+        self.r1.allowlist_node(self.fed1)
+        self.fed1.allowlist_node(self.r1)
+
+        self.r1.allowlist_node(self.r2)
+        self.r2.allowlist_node(self.r1)
+
+        self.r2.allowlist_node(self.sed2)
+        self.sed2.allowlist_node(self.r2)
+
+        self.r2.allowlist_node(self.r3)
+        self.r3.allowlist_node(self.r2)
+
+        self.r3.allowlist_node(self.fed3)
+        self.fed3.allowlist_node(self.r3)
+
+        self.r1.form(self.network_data, "router")
+        self.r1.permit_join(3600)
+        self.wait_for_completion(self.device_list)
+
+        self.logger.info(self.r1.ip6_lla)
+        self.logger.info(self.r1.ip6_thread_ula)
+
+        self.network_data.xpanid = self.r1.xpanid
+        self.network_data.panid = self.r1.panid
+
+        self.r2.join(self.network_data, "router")
+        self.wait_for_completion(self.device_list)
+
+        self.sed2.join(self.network_data, "sleepy-end-device")
+        self.sed2.set_sleep_poll_interval(POLL_INTERVAL)
+        self.wait_for_completion(self.device_list)
+
+        self.r3.join(self.network_data, "router")
+        self.wait_for_completion(self.device_list)
+
+        self.fed3.join(self.network_data, "end-node")
+        self.wait_for_completion(self.device_list)
+
+        self.fed1.join(self.network_data, "end-node")
+        self.wait_for_completion(self.device_list)
+
+    @testcase.test_method_decorator
+    def test02_verify_node_type_and_parent(self):
+        self.assertTrue(self.r1.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_LEADER)
+        self.assertTrue(self.r2.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_ROUTER)
+        self.assertTrue(self.r3.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_ROUTER)
+        self.assertTrue(self.fed1.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_END_DEVICE)
+        self.assertTrue(self.sed2.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_SLEEPY_END_DEVICE)
+        self.assertTrue(self.fed3.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_END_DEVICE)
+
+        r1_ext_address = self.r1.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+        fed1_parent = self.fed1.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        self.assertTrue(fed1_parent == r1_ext_address)
+
+        r2_ext_address = self.r2.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+        sed2_parent = self.sed2.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        self.assertTrue(sed2_parent == r2_ext_address)
+
+        r3_ext_address = self.r3.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+        fed3_parent = self.fed3.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        self.assertTrue(fed3_parent == r3_ext_address)
+
+    @testcase.test_method_decorator
+    def test03_verify_route_from_r1_to_r3(self):
+        # Wait till we have a valid "next hop" route on r1 towards r3
+        self.r1.add_prefix(PREFIX, stable=True, on_mesh=True, slaac=True, preferred=True)
+        self.wait_for_completion(self.device_list)
+
+        # Wait for prefix to get added to all the nodes
+        time.sleep(60)
```

Comment:
put it in a constant variable PREFIX_PROPAGATION_DELAY ?

Reply:
Ack